### PR TITLE
Feature/migrate cli tests

### PR DESCRIPTION
**Pull request by agduncan94**
* Deleted ClientIT, MockedIT, WDLWorkflowIT (moved all to CLI repository)
* Converted all tests in GeneralWorkflowIT to use API, moved old tests to CLI repository
* RegisterManual workflow now checks that the descriptor file extension is correct (previously only did this in UI and CLI)
Edit: Ended up rewriting BasicIT too.

https://github.com/dockstore/dockstore/issues/2970

### CODE REVIEW

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java  -65
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/dockstore/dockstore/pullRequest?prid=4563043&bid=15341580)

File: dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java [hunk 1]
```diff
@@ -161,6 +161,38 @@ public void resetDBBetweenTests() throws Exception {
...
+        assertEquals(Workflow.ModeEnum.STUB, workflow.getMode());
+
+        // Refresh
+        workflow = workflowsApi.refresh(workflow.getId());
+        assertEquals(Workflow.ModeEnum.FULL, workflow.getMode());
+
+        // Publish
+        if (toPublish) {
+            workflow = workflowsApi.publish(workflow.getId(), SwaggerUtility.createPublishRequest(true));
+            assertTrue(workflow.isIsPublished());
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java]: Trivial nitpick:
Could take this out of the if to make it more general
i.e.
`assertTrue(workflow.isIsPublished() == toPublish);`

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/dockstore/dockstore/pullRequest?prid=4563043&bid=15341580)

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/dockstore/dockstore/pullRequest?prid=4563043&bid=15341580)

**Comment by agduncan94** [comment]
@denis-yuen Requesting review again, added the BasicIT conversion and some small container API changes.

**Pull request merged**